lms/djangoapps/courseware/module_render.py

```diff
@@ -157,6 +157,7 @@ def toc_for_course(request, course, active_chapter, active_section, field_data_c
         for chapter in chapters:
             # Only show required content, if there is required content
             # chapter.hide_from_toc is read-only (boo)
+            name_without_spaces = chapter.display_name_with_default.replace(" ", "-").replace(":", "")
```

Comment:
@andy-armstrong I have looked at the ```slugify``` function that T&L team is trying to implement and it looks great to me. So I will follow up the PR and use that once https://github.com/edx/edx-platform/pull/8243 is merged.